src/http/h1/parse.rs

```diff
@@ -35,15 +36,24 @@ impl Http1Transaction for ServerTransaction {
                 trace!("Request.parse Complete({})", len);
                 let mut headers = Headers::with_capacity(req.headers.len());
                 let slice = buf.slice(len);
+                let path = req.path.unwrap();
+                let path_start = path.as_ptr() as usize -
+                    unsafe { MemStr::from_utf8_unchecked(slice.clone()) }.as_str()
```

Comment:
This doesn't need to clone the slice just to compare the pointer. It can be this:

```rust
let path_start = path.as_ptr() as usize - slice.as_ref().as_ptr() as usize;
```